python/ray/ml/train/integrations/sklearn/sklearn_trainer.py

```diff
@@ -0,0 +1,133 @@
+            )
+            result = trainer.fit()
+
+    Args:
+        estimator: A scikit-learn compatible estimator to use.
+        datasets: Ray Datasets to use for training and validation. Must include a
+            "train" key denoting the training dataset. If a ``preprocessor``
+            is provided and has not already been fit, it will be fit on the training
+            dataset. All datasets will be transformed by the ``preprocessor`` if
+            one is provided. All non-training datasets will be used as separate
+            validation sets, each reporting separate metrics.
+        label_column: Name of the label column. A column with this name
+            must be present in the training dataset.
+        scoring: Strategy to evaluate the performance of the model on
+            the validation sets and for cross-validation. Same as in
+            ``sklearn.model_selection.cross_validation``.
+            If ``scoring`` represents a single score, one can use:
+
+            - a single string;
+            - a callable that returns a single value.
+
+            If ``scoring`` represents multiple scores, one can use:
+
+            - a list or tuple of unique strings;
+            - a callable returning a dictionary where the keys are the metric
+            names and the values are the metric scores;
+            - a dictionary with metric names as keys and callables a values.
+
+        cv: Determines the cross-validation splitting strategy. If specified,
+            cross-validation will be ran on the train dataset, in addition to
+            computing metrics for validation datasets. Same as in
+            ``sklearn.model_selection.cross_validation``, with the exception of
+            None.
+            Possible inputs for ``cv`` are:
+
+            - None, to skip cross-validation.
+            - int, to specify the number of folds in a ``(Stratified)KFold``,
+            - ``CV splitter``,
+            - An iterable yielding (train, test) splits as arrays of indices.
+
+            For int/None inputs, if the estimator is a classifier and ``y`` is
+            either binary or multiclass, ``StratifiedKFold`` is used. In all
+            other cases, ``KFold`` is used. These splitters are instantiated
+            with ``shuffle=False`` so the splits will be the same across calls.
+        scaling_config: Configuration for how to scale data parallel training.
+        run_config: Configuration for the execution of the training run.
+        preprocessor: A ray.ml.preprocessor.Preprocessor to preprocess the
+            provided datasets.
+        resume_from_checkpoint: A checkpoint to resume training from.
+        **fit_params: Additional kwargs passed to ``estimator.fit()``
+            method.
+    """
+
+    def __init__(
+        self,
+        *,
+        estimator: BaseEstimator,
+        datasets: Dict[str, GenDataset],
+        label_column: Optional[str] = None,
```

Comment:
Just to clarify, label is optional to support e.g. unsupervised learning like clustering or so, correct?

Reply:
Yes!